Packs/ApiModules/Scripts/JSONFeedApiModule/JSONFeedApiModule.py

```diff
@@ -0,0 +1,163 @@
+                'fields': ['region', 'service']
+            }
+        """
+
+        self.source_name = source_name or 'JSON'
+        if feed_name_to_config:
+            self.feed_name_to_config = feed_name_to_config
+        else:
+            self.feed_name_to_config = {
+                self.source_name: {
+                    'url': url,
+                    'indicator': indicator or 'indicator',
+                    'extractor': extractor or '@',
+                    'fields': argToList(fields)
+                }}
+
+        # Request related attributes
+        self.url = url
+        self.verify = not insecure
+        if credentials:
+            self.auth = (credentials.get('username'), credentials.get('password'))
+        else:
+            self.auth = None
+
+        # Hidden params
+        self.headers = headers
+        self.cert = (cert_file, key_file) if cert_file and key_file else None
+
+    def build_iterator(self, **kwargs) -> List:
+        results = []
+        for feed_name, feed in self.feed_name_to_config.items():
+            r = requests.get(
+                url=feed.get('url'),
+                verify=self.verify,
+                auth=self.auth,
+                cert=self.cert,
+                headers=self.headers,
+                **kwargs
+            )
+
+            try:
+                r.raise_for_status()
+                data = r.json()
+                result = jmespath.search(expression=feed.get('extractor'), data=data)
+                results.append({feed_name: result})
+
+            except ValueError as VE:
+                raise ValueError(f'Could not parse returned data to Json. \n\nError massage: {VE}')
+
+        return results
+
+
+def test_module(client) -> str:
+    client.build_iterator()
+    return 'ok'
+
+
+def fetch_indicators_command(client: Client, indicator_type: str, update_context: bool = False,
+                             limit: int = None, feed_name: str = 'JSON', **kwargs) -> Union[Dict, List[Dict]]:
+    """
+    Fetches the indicators from client.
+    :param client: Client of a JSON Feed
+    :param indicator_type: the default indicator type
+    :param update_context: if *True* will also update the context with the indicators
+    :param limit: limits the number of context indicators to output
+    :param feed_name: The name of the feed.
+    """
+    indicators = []
+    for result in client.build_iterator(**kwargs):
+        for sub_feed_name, items in result.items():
+            feed_config = client.feed_name_to_config.get(sub_feed_name, client.source_name)
+            indicator_field = feed_config.get('indicator', 'indicator')
+            indicator_type = feed_config.get('indicator_type', indicator_type)
+            fields = feed_config.get('fields', [])
+            for item in items:
+                attributes = {'source_name': sub_feed_name}
+                attributes.update({f: item.get(f) for f in fields or item.keys() if f is not indicator_field})
```

Comment:
not sure i like this syntax, it's not clear enough

Reply:
@orenzohar